src/pfe/file-watcher/server/test/scripts/exec.sh

```diff
@@ -53,6 +53,42 @@ function downloadCwctl() {
     checkExitCode $? "Failed to give correct permission to run installer."
 }
 
+function setupInternalRegistryCredentials() {
+    # Adapted from https://github.com/eclipse/codewind-che-plugin/blob/master/scripts/che-setup.sh
+    echo -e "${BLUE}Setting up the Internal Registry Credentials ... ${RESET}"
+    OC_VERSION=$( oc version 2>&1 )
+    if [[ "$OC_VERSION" =~ "Client Version: version.Info{Major:\"4\"" ]]; then
+        REGISTRY_SECRET_ADDRESS="image-registry.openshift-image-registry.svc:5000"
+    else
+        REGISTRY_SECRET_ADDRESS="docker-registry.default.svc:5000"
+    fi
+
+    REGISTRY_SECRET_USERNAME=turbine-test-sa
+
+    oc delete sa $REGISTRY_SECRET_USERNAME
+    oc create sa $REGISTRY_SECRET_USERNAME
+
+    oc policy add-role-to-user system:image-builder system:serviceaccount:$NAMESPACE:$REGISTRY_SECRET_USERNAME
+
+    ENCODED_TOKEN=$(oc get secret $(oc describe sa $REGISTRY_SECRET_USERNAME | tail -n 2 | head -n 1 | awk '{$1=$1};1') -o json | jq ".data.token")
+    REGISTRY_SECRET_PASSWORD=$( node ./scripts/utils.js decode $ENCODED_TOKEN )
+}
+
+function setupRegistrySecret() {
+    if [[ ! -z $INTERNAL_REGISTRY ]]; then
+        setupInternalRegistryCredentials
```

Comment:
So if internal registry is passed, we will set the default registry as well the user registry? I think we should exit out of the function if internal registry was passed

Reply:
we cant exit, `setupInternalRegistryCredentials`just  sets the variables for address, username and password with the logic from che setup script

The rest of the function `setupRegistrySecret` (diff function, look at the name) calls the curl in PFE. So basically `setupInternalRegistryCredentials` just overwrites the variables before calling curl

Reply:
Oh, I see. Ok then its fine